src/Libraries/RevitNodes/Elements/AdaptiveComponent.cs

```diff
@@ -225,100 +54,24 @@ private void InitAdaptiveComponent(Autodesk.Revit.DB.FamilyInstance familyInstan
         #region Internal mutators
 
         /// <summary>
-        /// Set the family symbol for the internal family instance 
-        /// </summary>
-        /// <param name="ft"></param>
-        private void InternalSetFamilySymbol(FamilyType ft)
-        {
-            TransactionManager.Instance.EnsureInTransaction(Document);
-
-            InternalFamilyInstance.Symbol = ft.InternalFamilySymbol;
-
-            TransactionManager.Instance.TransactionTaskDone();
-
-        }
-
-        /// <summary>
-        /// Set the positions of the internal family instance from a list of XYZ points
+        /// Update the placement points of the adaptive component instance to the given points
         /// </summary>
-        /// <param name="points"></param>
-        private void InternalSetPositions( XYZ[] points )
+        /// <param name="fi"></param>
+        /// <param name="pnt"></param>
+        private static void UpdatePlacementPoints(Autodesk.Revit.DB.FamilyInstance fi, XYZ[] pnts)
         {
-            TransactionManager.Instance.EnsureInTransaction(Document);
-
-            IList<ElementId> placePointIds = AdaptiveComponentInstanceUtils.GetInstancePlacementPointElementRefIds(InternalFamilyInstance);
+            IList<ElementId> placePointIds = AdaptiveComponentInstanceUtils.GetInstancePlacementPointElementRefIds(fi);
 
-            if (placePointIds.Count() != points.Count())
+            if (placePointIds.Count() != pnts.Count())
                 throw new Exception(Properties.Resources.InputPointParamsMismatch);
 
-            // Set the position of each placement point
             int i = 0;
             foreach (var id in placePointIds)
```

Comment:
It's better to run a loop over i=0 to i < pnts.Count() so that you don't have to think about increment of i inside the loop.